engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/br/CNPJValidator.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.constraintvalidators.hv.br;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import org.hibernate.validator.constraints.Mod11Check;
+import org.hibernate.validator.constraints.br.CNPJ;
+import org.hibernate.validator.internal.constraintvalidators.hv.Mod11CheckValidator;
+import org.hibernate.validator.internal.util.annotationfactory.AnnotationDescriptor;
+import org.hibernate.validator.internal.util.annotationfactory.AnnotationFactory;
+
+/**
+ * @author Hardy Ferentschik
+ */
+public class CNPJValidator implements ConstraintValidator<CNPJ, CharSequence> {
+	private final Mod11CheckValidator firstMod11CheckWithSeparator = new Mod11CheckValidator();
```

Comment:
I'm not quite sure why there are four validators needed? And were do the numbers used for their initialization come from? Some short explanation of this enitre scheme would be helpful.